### ISSUE TITLE
Input text cursor rendering issues

### ISSUE THREAD

**Issue by tpecholt** [ocornut/imgui#7031]
**Version/Branch of Dear ImGui:**

Version: 1.90
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_android.cpp + imgui_impl_opengl3.cpp
Compiler: Android NDK Clang
Operating System: Android

**My Issue/Question:**

I would like the rendering of the text cursor in input controls to be more configurable. 

1. Right now the rendering is done by drawing a 1px line. That is problematic with DPI scaling so the cursor is barely visible. 

2. The color should be configurable too currently it is fixed to ImGuiCol_Text. Android uses different color for text cursor and text.

**Screenshots/Video**

![cursor](https://github.com/ocornut/imgui/assets/1256102/035268a9-5b73-437f-8be7-787ff00b2e94)

**Comment by ocornut** [comment]
We don't have a standardized mechanism to specify general/global scaling and would probably wait until we have one.

Similarly I don't disagree we could have a specific color for it but it likely would have to wait until we update the style system to avoid cluttering out existing list.

**Comment by tope99** [comment]
Yes but the difference is for other settings like WindowPadding, ItemSpacing... I can call ImStyle::ScaleAllSizes(). And as for fonts I can load them with scaled size. So using existing functionality I am already able to achieve DPI scaling of UI elements. But the width of the text cursor is fixed so I can't scale it unless I directly modify ImGui code.

**Comment by tksuoran** [comment]
I would also like to have configurable cursor color.

**Comment by ocornut** [comment]
For scaling, I'm currently working on general scaling system and standardized scaling factor so I would like this to be part of it and not another style variable. But I could add the color sooner.

**Comment by ocornut** [comment]
I've added a color for the cursor.
I cannot add the thickness scale yet because the way this will be exposed depends on questions I don't have the right answer just yet (but actively working on it).

**Comment by tksuoran** [comment]
Awesome - Thank you!

**Comment by ocornut** [comment]
I have pushed a basic integer-rounded scale value for cursor thickness, based on the value passed to `style.ScaleAllSizes()`.
This is not perfect but I haven't had time to standardize style scaling technique yet.

**Comment by ocornut** [comment]
Added `style.InputTextCursorSize` in 24a80f7.
